web-frontend/src/pages/TransferCoins/ReceiveCoins/ReceiveCoins.tsx

```diff
@@ -23,19 +24,47 @@ function ReceiveCoins() {
   const formattedAddress = maskAddress(address);
   const VITE_BASE_APP = import.meta.env.VITE_BASE_APP;
   const baseURL = window.location.origin;
-
+  const [modal, setModal] = useState(false);
   const presetURL = `${baseURL}${VITE_BASE_APP}/send-redirect/?to=${address}`;
+  const [url, setURL] = useState(presetURL);
+  const [provider, setProvider] = useState('');
+  const [amount, setAmount] = useState('');
+
+  console.log('url', url);
+
+  const handleGenerate = () => {
+    const amountArg = amount ? `&amount=${amount}` : '';
+    const providerArg = provider ? `&provider=${provider}` : '';
+    const newURL = presetURL + amountArg + providerArg;
+    setURL(newURL);
+  };
+
+  const modalArgs = {
+    account,
+    modal,
+    url,
+    setModal,
+    setAmount,
+    setProvider,
+    handleGenerate,
+  };
+
+  const copyContentArgs = {
+    content: address,
+    formattedContent: formattedAddress,
+  };
 
   return (
     <div className="flex flex-row items-center gap-5">
-      <QRCodeReact value={presetURL} size={165} />
+      <QRCodeReact value={url} size={165} />
       <div className="flex flex-col w-full gap-3.5">
         <p>{Intl.t('receive.account-address')}</p>
-        <CopyAddress address={address} formattedAddress={formattedAddress} />
-        <Button preIcon={<FiLink size={24} />}>
+        <CopyContent {...copyContentArgs} />
+        <Button onClick={() => setModal(!modal)} preIcon={<FiLink size={24} />}>
           {Intl.t('receive.account-receive')}
         </Button>
       </div>
+      {modal ? <GenerateLink {...modalArgs} /> : null}
```

Comment:
OK, now I understand why you pass modal to GenerateLink... maybe there is something better to do?

Reply:
I just set it to false directly  in the modal so I just need setModal now